File: CPP/PSU.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

// Kelas Psu yang mewarisi dari kelas Komponen
class Psu : public Komponen {
    private:
        int watt;
        string sertifikasi;
    
    public:
        Psu() {}

        // Konstruktor dengan parameter untuk menginisialisasi watt, sertifikasi, merk, dan nama
        Psu(int watt, string sertifikasi, string merk, string nama) 
            : Komponen(merk, nama) {
            this->watt = watt;
            this->sertifikasi = sertifikasi;
        }
    

        // Setter
        void setWatt(int watt) {
            this->watt = watt;
        }
    
        void setSertifikasi(string sertifikasi) {
            this->sertifikasi = sertifikasi;
        }
    
        // Getter
        int getWatt() {
            return watt;
        }
    
        string getSertifikasi() {
            return sertifikasi;
        }
    
    ~Psu()
    {
        
    }
};

File: python/Psu.py
from Komponen import Komponen

class Psu(Komponen):
    #atribut private
    __watt = ""
    __sertifikasi = ""

    #constructor
    def __init__(self, watt, sertifikasi, merk, nama):
        super().__init__(merk, nama)
        self._watt = watt
        self._sertifikasi = sertifikasi

    #setter
    def set_watt(self, watt):
        self._watt = watt

    def set_sertifikasi(self, sertifikasi):
        self._sertifikasi = sertifikasi

    #getter
    def get_watt(self):
        return self._watt

    def get_sertifikasi(self):
        return self._sertifikasi